services/src/main/java/org/keycloak/authentication/requiredactions/TermsAndConditions.java

```diff
@@ -46,18 +50,22 @@ public void evaluateTriggers(RequiredActionContext context) {
 
     @Override
     public void requiredActionChallenge(RequiredActionContext context) {
-        Response challenge =  context.form().createForm("terms.ftl");
+        Response challenge = context.form().createForm("terms.ftl");
         context.challenge(challenge);
     }
 
     @Override
     public void processAction(RequiredActionContext context) {
         if (context.getHttpRequest().getDecodedFormParameters().containsKey("cancel")) {
+            context.getUser().removeAttribute(USER_ATTRIBUTE);
             context.failure();
             return;
         }
-        context.success();
 
+        SimpleDateFormat dateTimeFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
+        context.getUser().setAttribute(USER_ATTRIBUTE, Arrays.asList(dateTimeFormat.format(new Date())));
```

Comment:
We store time as seconds from 1970 everywhere else, as well as UserModel.getCreatedTimestamp so it should be changed to that to make it consistent (use org.keycloak.common.util.Time.currentTime)